src/Main/Main.cc

```diff
@@ -86,6 +86,10 @@ int main(int argc, char* argv[]) {
             }
         }
 
+        if (settings.controller_build) {
+            setenv("CARTA_USER_DEPLOYMENT", "controller", 1);
```

Comment:
This should not be reusing (and overwriting) the same environment variable that is being used to detect a docker deployment. These should be saved to separate telemetry entries.

It should also not be setting an environment variable here at all -- this value should be saved internally in a variable which can then be used in the `Session` code -- I suggest setting a static variable on the session class (you can see how this is done for the exit timeouts in `Main.cc`).